docs/build-with-powerloom/devnet/getting-started.md

```diff
@@ -76,26 +86,170 @@ Clone the repository against the respective branch (main by default). Open the t
+- `SIGNER_ACCOUNT_PRIVATE_KEY`: The private key corresponding to the burner wallet address
+- `SLOT_ID`: Enter your Devnet Slot ID. [You can find the slotID by visiting the [Devnet Dashboard](https://devnet-mint.powerloom.dev/dashboard), or on the [Sepolia block explorer](https://sepolia.etherscan.io/) by searching for the transaction hash of your node mint]
+- `PROTOCOL_STATE_CONTRACT`: Choose the contract address for the protocol state with respect to the Epoch size.
 
-Our system leverages the Git submodule architecture to seamlessly manage these components. To dive deeper into how these elements integrate and function within our larger system, check out the [architecture documentation](https://docs.powerloom.io/docs/build-with-powerloom/snapshotter-node/architecture).
+:::note
+Currently, there are two variations of Protocol State Contracts available. One where [Epoch](../../Protocol/Specifications/Epoch.md) size is set to 1 and other where epoch size is set to 10 blocks. 
 
-To optimize the development process, we recommend forking the templates for:
+You can find the latest Epoch Contracts in the [example.env](https://github.com/PowerLoom/deploy/blob/devnet/env.example) file. 
 
-- Snapshotter Configs: https://github.com/PowerLoom/snapshotter-configs
-- Snapshotter Computes: https://github.com/PowerLoom/snapshotter-computes
+This should allow developers to build and experiment with a variety of use cases. If you have any custom needs, reach out to the [team](https://discord.com/invite/powerloom)
+:::
+
+**Already Given:**  
+
+- `SNAPSHOT_CONFIG_REPO`: https://github.com/powerloom/snapshotter-configs
+- `SNAPSHOT_CONFIG_REPO_BRANCH`: devnet
+- `SNAPSHOTTER_COMPUTE_REPO`: https://github.com/powerloom/snapshotter-computes
+- `SNAPSHOTTER_COMPUTE_REPO_BRANCH`: devnet
+- `PROST_RPC_URL`: The URL for the Powerloom Protocol Chain RPC service.
+- `RELAYER_HOST`: The relayer URL for the Powerloom Protocol Chain.
+- `NAMESPACE`: The unique key used to identify your project namespace.
+- `POWERLOOM_REPORTING_URL`: The URL for reporting to PowerLoom.
+- `PROST_CHAIN_ID`: The Devnet chain ID
+
+**Optional variables:**
+
+- `IPFS_URL`: The URL for the IPFS (InterPlanetary File System) service in HTTP(s) (e.g., `https://ipfs.infura.io:5001`) or multiaddr format (e.g., `/dns4/ipfs.infura.io/tcp/5001/https`).
+- `IPFS_API_KEY`: The API key for the IPFS service (if required).
+- `IPFS_API_SECRET`: The API secret for the IPFS service (if required).
+- `SLACK_REPORTING_URL`: The URL for reporting to Slack.
+
+#### Step 4: Set Up the Codebase
+
+Set up the codebase by running the `bootstrap.sh` command in the terminal:
+
+```bash
```

Comment:
The fine print here should be that depending upon the shell used by the developer, this should change. `bash` is just one shell out of many that a developer may use. It does come built in with most linux and macOS distributions, hence its popularity. `zsh` is something and quite a few on our team use it, and it is one of the most popular ones among developers across the world.